test/powershell/engine/Basic/ValidateAttributes.Tests.ps1

```diff
@@ -260,4 +260,124 @@ Describe 'Validate Attributes Tests' -Tags 'CI' {
                 $ScriptBlock | Should Not Throw
         }
     }
+
+    Context "ValidateNotNull, ValidateNotNullOrEmpty and Not-Null-Or-Empty check for Mandatory parameter" {
+
+        BeforeAll {
+            function MandType {
+                param(
+                    [Parameter(Mandatory, ParameterSetName = "ByteArray")]
+                    [byte[]] $ByteArray,
+
+                    [Parameter(Mandatory, ParameterSetName = "ByteList")]
+                    [System.Collections.Generic.List[byte]] $ByteList,
+
+                    [Parameter(Mandatory, ParameterSetName = "ByteCollection")]
+                    [System.Collections.ObjectModel.Collection[byte]] $ByteCollection,
+
+                    [Parameter(ParameterSetName = "Default")]
+                    $Value
+                )
+            }
+
+            function NotNullFunc {
+                param(
+                    [ValidateNotNull()]
+                    $Value,
+                    [string] $TestType
+                )
+
+                switch ($TestType) {
+                    "COM-Enumerable" { $Value | ForEach-Object Name }
+                    "Enumerator"     {
+                        $items = foreach ($i in $Value) { $i }
+                        $items -join ","
+                    }
+                }
+            }
+
+            function NotNullOrEmptyFunc {
+                param(
+                    [ValidateNotNullOrEmpty()]
+                    $Value,
+                    [string] $TestType
```

Comment:
Can you add a parameter like: `[System.Nullable``1[[System.Int32]]] $nullableInt`

Reply:
A `[System.Nullable[int]]` parameter is not useful in this case because the type is not a collection. Hence `ValidateNotNullOrEmpty` attribute won't validate it as a collection.